src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1255,6 +1263,279 @@ enum H : C { }
+                // (5,5): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
+                //     public data int B = 0;
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "public data int B = 0;").WithArguments("System.Runtime.CompilerServices.IsExternalInit").WithLocation(5, 5),
+                // (5,5): error CS0106: The modifier 'public' is not valid for this item
+                //     public data int B = 0;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "public data int B = 0;").WithArguments("public").WithLocation(5, 5),
+                // (5,12): error CS8652: The feature 'records' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public data int B = 0;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "data").WithArguments("records").WithLocation(5, 12),
+                // (6,5): error CS0246: The type or namespace name 'data' could not be found (are you missing a using directive or an assembly reference?)
+                //     data C;
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "data").WithArguments("data").WithLocation(6, 5),
+                // (6,10): warning CS0169: The field 'X.C' is never used
+                //     data C;
+                Diagnostic(ErrorCode.WRN_UnreferencedField, "C").WithArguments("X.C").WithLocation(6, 10)
+            );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void DataProperties1([CombinatorialValues("class", "struct", "record")] string typeKind)
+        {
+            var src = @$"
+{typeKind} X
+{{
+    data int A;
+    public data int B;
+    readonly data int C;
+    static data int D;
+    data int E;
+}}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (5,5): error CS0106: The modifier 'public' is not valid for this item
+                //     public data int B;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "public data int B;").WithArguments("public").WithLocation(5, 5),
+                // (6,5): error CS0106: The modifier 'readonly' is not valid for this item
+                //     readonly data int C;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "readonly data int C;").WithArguments("readonly").WithLocation(6, 5),
+                // (7,5): error CS0106: The modifier 'static' is not valid for this item
+                //     static data int D;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "static data int D;").WithArguments("static").WithLocation(7, 5)
+            );
+        }
+
+        [Fact]
+        public void DataProperties2()
+        {
+            var src = @"
+class C
+{
+    data int X = 0;
+    data int Y = """";
```

Comment:
Regarding initialization values:

We should probably add a nullability test (or file issue): `data string Z = null;`
Also, include a more complex expression in initializer, verifying what's in scope or not (record parameters, out vars, ....).
Calling an unsafe method for initialization.
Verifying semantic model and IOperation on initializer expression (or file issue).

Reply:
See `DataProperties3` which verifies definite assignment and scoping through diagnostics.

Reply:
ping

---
In reply to: [441124214](https://github.com/dotnet/roslyn/pull/44883#discussion_r441124214) [](ancestors = 441124214)

Reply:
Things that I don't have:

- IOperation test
- test that we get events for this symbol, including symbol declaration and events for the initializer

I'll add those to the test plan to avoid bloating this PR too much.